### ISSUE TITLE
Vagrant does not shutdown Virtual Box Headless Frontend upon reload or halt

### ISSUE THREAD

**Issue by whannes** [hashicorp/vagrant#12520]
### Vagrant version
Vagrant 2.2.18

### Host operating system
OS Name:                   Microsoft Windows 10 Pro
OS Version:                10.0.19043 N/A Build 19043
OS Manufacturer:           Microsoft Corporation
OS Configuration:          Standalone Workstation
OS Build Type:             Multiprocessor Free
Processor(s):              1 Processor(s) Installed.
                           [01]: Intel64 Family 6 Model 165 Stepping 2 GenuineIntel ~2592 Mhz
Total Physical Memory:     32'402 MB

### VirtualBox
VirtualBox 6.1.26 r145957 

### Guest operating system
CentOS 7

### Vagrantfile

```ruby
# -*- mode: ruby -*-
# vi: set ft=ruby :

Vagrant.configure("2") do |config|
	config.vm.box = "generic/centos7"
	config.vbguest.auto_update = false
	config.vbguest.auto_reboot  = false
end
```
### Debug output

On my machine it seems as if the process `VirtualBox Headless Frontend` started with `vagrant up` is not properly closed upon `vagrant halt`  or `vagrant reload`. Without manually killing these processes I get the following behaviour

```
C:\XY>vagrant halt
==> default: Attempting graceful shutdown of VM...

C:\Developer\vagrant\centos7a>vagrant up
Bringing machine 'default' up with 'virtualbox' provider...
==> default: Checking if box 'generic/centos7' version '3.4.2' is up to date...
==> default: Clearing any previously set forwarded ports...
==> default: Clearing any previously set network interfaces...
==> default: Preparing network interfaces based on configuration...
    default: Adapter 1: nat
==> default: Forwarding ports...
    default: 22 (guest) => 2222 (host) (adapter 1)
==> default: Running 'pre-boot' VM customizations...
==> default: Booting VM...
There was an error while executing `VBoxManage`, a CLI used by Vagrant
for controlling VirtualBox. The command and stderr is shown below.

Command: ["startvm", "7b0b7a06-540b-47e1-b1e0-f9ad35b007ca", "--type", "headless"]

Stderr: VBoxManage.exe: error: The VM session was closed before any attempt to power it on
VBoxManage.exe: error: Details: code E_FAIL (0x80004005), component SessionMachine, interface ISession
```

### Actual behavior

`vagrant reload` cannot be used. There is a workaround to reload the machine using `vagrant halt` and `vagrant up`: If I manually kill all `VirtualBox Headless Frontend` processes after `vagrant halt` but before `vagrant up` it works as expected:

![grafik](https://user-images.githubusercontent.com/41183585/133438531-db586d0d-9e4b-4b82-be90-5a55b15230da.png)

```
C:\XY>vagrant halt
==> default: Attempting graceful shutdown of VM...

C:\XY>vagrant up
Bringing machine 'default' up with 'virtualbox' provider...
==> default: Checking if box 'generic/centos7' version '3.4.2' is up to date...
==> default: Clearing any previously set forwarded ports...
==> default: Clearing any previously set network interfaces...
==> default: Preparing network interfaces based on configuration...
    default: Adapter 1: nat
==> default: Forwarding ports...
    default: 22 (guest) => 2222 (host) (adapter 1)
==> default: Running 'pre-boot' VM customizations...
==> default: Booting VM...
==> default: Waiting for machine to boot. This may take a few minutes...
    default: SSH address: 127.0.0.1:2222
    default: SSH username: vagrant
    default: SSH auth method: private key
==> default: Machine booted and ready!
==> default: Checking for guest additions in VM...
    default: The guest additions on this VM do not match the installed version of
    default: VirtualBox! In most cases this is fine, but in rare cases it can
    default: prevent things such as shared folders from working properly. If you see
    default: shared folder errors, please make sure the guest additions within the
    default: virtual machine match the version of VirtualBox you have installed on
    default: your host and reload your VM.
    default:
    default: Guest Additions Version: 5.2.44
    default: VirtualBox Version: 6.1
==> default: Machine already provisioned. Run `vagrant provision` or use the `--provision`
==> default: flag to force provisioning. Provisioners marked to run always will still run.
```

### Steps to reproduce

1. `vagrant up`
2. `vagrant halt` 
3. `vagrant up`

or

1. `vagrant up`
2. `vagrant reload`

**Comment by PaulNeumann** [comment]
This appears to be the same issue as #12484. Other than manually stopping the VBoxHeadless processes, the only workaround I've found is downgrading to VirtualBox 6.1.24.

**Comment by elb98rm** [comment]
Can confirm this behaviour AND workaround with my own setup. Can post info if required for debugging.

**Comment by PaulNeumann** [comment]
This issue seems to be resolved in VirtualBox 6.1.28. See [this VirtualBox ticket](https://www.virtualbox.org/ticket/20574) and [the changelog for VirtualBox 6.1.28](https://www.virtualbox.org/wiki/Changelog).

From the VirtualBox 6.1.28 changelog: "Windows Host: Fixed VBoxHeadless process sticking around after VM is closed (bug #20574)".

**Comment by chrisroberts** [comment]
Closing as this was a VirtualBox issue.
